hibernate-core/src/main/java/org/hibernate/spi/NavigablePath.java

```diff
@@ -140,9 +140,10 @@ public boolean equals(Object other) {
 			if ( ! Objects.equals( getAlias(), otherNavigablePath.getAlias() ) ) {
 				return false;
 			}
+			return Objects.equals( getRealParent(), otherNavigablePath.getRealParent() );
 		}
 
-		return Objects.equals( getParent(), otherPath.getParent() );
+		return Objects.equals( getRealParent(), otherPath.getParent() );
```

Comment:
probably better to revert this to return Objects.equals( getParent(), otherPath.getParent() );